### PR TITLE
[INLONG-7194][DataProxy] Add index log statistics for MQ sink

### LINKED ISSUE #7194: [Improve][DataProxy] Migrate index log statistics for the new mq layer
### Description

Add index log for new sink layer.

### InLong Component

InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by woofyzhao**
- Fixes #7194

### Motivation

Missing old index log statistics for simple events. Migrate the old logic from pulsar sink and tube sink.

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java [hunk 1]
```diff
@@ -268,4 +308,40 @@ public void processSendFail(BatchPackProfile currentRecord, String topic, long s
...
+                .append(getSinkName()).append(SEP_HASHTAG).append(topic)
+                .append(SEP_HASHTAG).append(streamId).append(SEP_HASHTAG)
+                .append(nodeIp).append(SEP_HASHTAG).append(NetworkUtils.getLocalIp())
+                .append(SEP_HASHTAG).append(evenProcType.getRight()).append(SEP_HASHTAG)
+                .append(DateTimeUtils.ms2yyyyMMddHHmm(dataTimeL));
+        // count data
+        if (isSuccess) {
+            monitorIndex.addAndGet(newBase.toString(),
+                    intMsgCnt, 1, event.getBody().length, 0);
+            monitorIndexExt.incrementAndGet("PULSAR_SINK_SUCCESS");
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java:340]: MQ must not be pulsar.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java [hunk 2]
```diff
@@ -52,6 +68,10 @@ public class MessageQueueZoneSinkContext extends SinkContext {
     private final CacheClusterConfigHolder cacheHolder;
     private final INLONG_COMPRESSED_TYPE compressType;
 
+    private final MQClusterConfig mqConfig;
+    private MonitorIndex monitorIndex;
+    private MonitorIndexExt monitorIndexExt;
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java:73]: No code use the data of monitorIndex and monitorIndexExt.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java [hunk 3]
```diff
@@ -268,4 +308,40 @@ public void processSendFail(BatchPackProfile currentRecord, String topic, long s
...
+
+    private void appendIndex(Event event, boolean isSuccess) {
+        if (event == null) {
+            return;
+        }
+        if (mqConfig.getStatIntervalSec() <= 0) {
+            return;
+        }
+        // add monitor items base file storage
+        String topic = event.getHeaders().get(ConfigConstants.TOPIC_KEY);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java:320]: These logic can implement in the user-defined MQ Handler.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java (hunk 2 above)
Comment (woofyzhao) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java:73]: It's intended to print statistics to file, index.log.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java (hunk 3 above)
Comment (woofyzhao) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java:320]: This is not MQ specific but message protocol specific, which is handled by common context logic.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java (hunk 1 above)
Comment (woofyzhao) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/sink/mq/MessageQueueZoneSinkContext.java]: Fixed

**Comment by woofyzhao** [comment]
Consider using file listener instead. Close this for now.

**Pull request closed without merging**